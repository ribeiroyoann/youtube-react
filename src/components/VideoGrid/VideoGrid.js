import React from "react";
import "./VideoGrid.scss";
import { Divider } from "semantic-ui-react";
import { VideoPreview } from "../VideoPreview/VideoPreview";
import { VideoGridHeader } from "./VideoGridHeader/VideoGridHeader";

// PLACEHOLDER BEFORE API

// export function VideoGrid(props) {
//   const divider = props.hideDivider ? null : <Divider />;
//   return (
//     <React.Fragment>
//       <VideoGridHeader title={props.title} />
//       <div className="video-grid">
//         <VideoPreview />
//         <VideoPreview />
//         <VideoPreview />
//         <VideoPreview />
//         <VideoPreview />
//         <VideoPreview />
//         <VideoPreview />
//         <VideoPreview />
//         <VideoPreview />
//         <VideoPreview />
//         <VideoPreview />
//         <VideoPreview />
//       </div>
//       {divider}
//     </React.Fragment>
//   );
// }

export function VideoGrid(props) {
  if (!props.videos || !props.videos.length) {
    return <div />;
  }
  const gridItems = props.videos.map(video => {
    return <VideoPreview video={video} key={video.id} />;
  });

  const divider = props.hideDivider ? null : <Divider />;
  return (
    <React.Fragment>
      <VideoGridHeader title={props.title} />
      <div className="video-grid">{gridItems}</div>
      {divider}
    </React.Fragment>
  );
}
