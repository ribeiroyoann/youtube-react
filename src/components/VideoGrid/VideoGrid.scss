.video-grid {
  display: flex;
  flex-wrap: wrap;
  //   justify-content: space-around;

  & > * {
    margin-left: 4px;
    margin-bottom: 24px;
  }
}
