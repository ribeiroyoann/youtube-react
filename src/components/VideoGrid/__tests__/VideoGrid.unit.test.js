import React from "react";
import { shallow } from "enzyme";
import { VideoGrid } from "../VideoGrid";

describe("VideoGrid", () => {
  test("renders VideoGrid no props", () => {
    const wrapper = shallow(<VideoGrid />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders VideoGrid with title", () => {
    const wrapper = shallow(<VideoGrid title="title" />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders VideoGrid no divider", () => {
    const wrapper = shallow(<VideoGrid hideDivider={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
