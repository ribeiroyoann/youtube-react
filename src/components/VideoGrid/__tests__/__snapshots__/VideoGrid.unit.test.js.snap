// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid renders VideoGrid no divider 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
</Fragment>
`;

exports[`VideoGrid renders VideoGrid no props 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders VideoGrid with title 1`] = `
<Fragment>
  <VideoGridHeader
    title="title"
  />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;
